File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/RefreshTokenRepository.java
package com.ll.hotel.domain.member.member.repository;

import com.ll.hotel.global.jwt.dto.RefreshToken;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface RefreshTokenRepository extends CrudRepository<RefreshToken, String> {
    Optional<RefreshToken> findByRefreshToken(String refreshToken);
    Optional<RefreshToken> findByAccessToken(String accessToken);
    boolean existsByRefreshToken(String refreshToken);
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/repository/RefreshTokenRepository.kt
package com.ll.hotel.domain.member.member.repository

import com.ll.hotel.global.jwt.dto.RefreshToken
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface RefreshTokenRepository : CrudRepository<RefreshToken, String> {
    fun findByRefreshToken(refreshToken: String): Optional<RefreshToken>
    fun findByAccessToken(accessToken: String): Optional<RefreshToken>
    fun existsByRefreshToken(refreshToken: String): Boolean
} 